src/api/spec/factories/download_repository_factory.rb

```diff
@@ -6,16 +6,10 @@
     repository { association :repository, architectures: [arch] }
 
     before(:create) do |download_repository|
-      repo_arch = RepositoryArchitecture.find_by(
+      RepositoryArchitecture.find_or_create_by!(
```

Comment:
> A RepositoryArchitecture will allways be created on an existing Architecture.

Then creating it is not really needed, right?

Reply:
Actually, removing the `_create_by!` part will result in only finding a record, and not creating or updating any other record in the database.

Do you mean we could get rid of the `before(:create) do` block, also?

Reply:
I mean that in the PR description you say `Get rid of unnecessary code. A RepositoryArchitecture will always be created on an existing Architecture.` 
The `create(:repository_architecture` that you removed, is now is the `_create_by!` part, but then I don't understand the part regarding the `Get rid of unnecessary code`... so: I understand you are refactoring not actually removing code, then?

Reply:
I guess the important part is "existing Architecture".

Let me explain it in two steps.

First, I removed the code where an Architecture is created with `Architecture.find_or_create_by!(name: download_repository.arch)`. That's because we will always add `RepositoryArchitectures` over already existing architectures. That is what this sentence is about: `A RepositoryArchitecture will always be created on an existing Architecture.`

Secondly, I removed the comments and the `unless repo_arch` block after the comments, joining the creation of not found records in `RepositoryArchitectures` in a `RepositoryArchitecture.find_or_create_by!(` call.

`Get rid of unnecessary code` applies to the sentences of the comments. We no longer need the separation in two steps.